pkg/deviceclaimingserver/enddevices/ttjsv2/ttjs.go

```diff
@@ -177,10 +178,13 @@ func (c *TTJS) Claim(ctx context.Context, joinEUI, devEUI types.EUI64, claimAuth
 
 	switch resp.StatusCode {
 	case http.StatusBadRequest:
-		return errBadRequest.WithAttributes("message", errResp.Message)
+		return errInternalError.WithAttributes("message", errResp.Message)
 	case http.StatusNotFound:
 		return errDeviceNotProvisioned.WithAttributes("dev_eui", devEUI)
 	case http.StatusForbidden:
+		if errResp.Message == "claim failed with given owner token" {
+			return errBadRequest.WithAttributes("message", errResp.Message)
+		}
 		return errDeviceAccessDenied.WithAttributes("dev_eui", devEUI)
```

Comment:
In the case that the "claim failed with given owner token", that is when we want to convey a 400 with an informative message to the user.

So while the error code is good (400), you should basically return `errDeviceAccessDenied` that has error code 400 (not 500).

If it returns 403 but the message is different, return `errInternalError`.

Essentially, there is no `errBadRequest` here anymore. You can remove it. Run `tools/bin/mage go:messages js:translations` to get rid of the messages everywhere (otherwise CI will complain).

Reply:
I've changed it and now it returns `errDeviceAccessDenied` with error code 400 when the message is "claim failed with given owner token" and with `errInternalError` in other cases.

There is still one place where we use `errBadRequest` when the batch unclaim list is empty https://github.com/TheThingsNetwork/lorawan-stack/blob/e2b9df1f1c3e1a8ecaefc4fc15a8640f9fc5f13b/pkg/deviceclaimingserver/enddevices/ttjsv2/ttjs.go#L338

I've added the new `errInternalError` to the translations.

Reply:
OK, this is good now.